tfworker/commands/terraform.py

```diff
@@ -0,0 +1,485 @@
+                self._backend,
+            )
+            # run terraform init
+            try:
+                self._run(definition, "init", debug=self._show_output)
+            except TerraformError:
+                click.secho("error running terraform init", fg="red")
+                raise SystemExit(1)
+
+            click.secho("planning definition: {}".format(definition.tag), fg="green")
+
+            # run terraform plan
+            try:
+                self._run(
+                    definition,
+                    "plan",
+                    debug=self._show_output,
+                    plan_action=self._plan_for,
+                )
+            except PlanChange:
+                execute = True
+            except TerraformError:
+                click.secho(
+                    "error planning terraform definition: {}!".format(definition.tag),
+                    fg="red",
+                )
+                raise SystemExit(1)
+
+            if self._force_apply:
+                execute = True
+
+            if execute and self._tf_apply:
+                if self._force_apply:
+                    click.secho(
+                        f"force apply for {definition.tag}, applying",
+                        fg="yellow",
+                    )
+                else:
+                    click.secho(
+                        f"plan changes for {definition.tag}, applying",
+                        fg="yellow",
+                    )
+            elif execute and self._destroy:
+                click.secho(
+                    f"plan changes for {definition.tag}, destroying",
+                    fg="yellow",
+                )
+            elif not execute:
+                click.secho(f"no plan changes for {definition.tag}", fg="yellow")
+                continue
+
+            try:
+                self._run(
+                    definition,
+                    self._plan_for,
+                    debug=self._show_output,
+                )
+            except TerraformError:
+                click.secho(
+                    f"error with terraform {self._plan_for} on definition"
+                    f" {definition.tag}, exiting",
+                    fg="red",
+                )
+                raise SystemExit(1)
+            else:
+                click.secho(
+                    f"terraform {self._plan_for} complete for {definition.tag}",
+                    fg="green",
+                )
+
+    def _run(self, definition, command, debug=False, plan_action="init"):
+        """Run terraform."""
+        params = {
+            "init": "-input=false -no-color",
+            "plan": "-input=false -detailed-exitcode -no-color",
+            "apply": "-input=false -no-color -auto-approve",
+            "destroy": "-input=false -no-color -force",
+        }
+
+        if plan_action == "destroy":
+            params["plan"] += " -destroy"
+
+        env = os.environ.copy()
+        # TODO(jwiles): Is this safe?
```

Comment:
this is safe within the worker, but could leak something to hooks, if we want to ensure safety for hooks shlex.quote() all the return values from the authenticator.env's where env vars are set.